### PR TITLE
Feature: choose covariance method

### PR DESCRIPTION
**Pull request by vrognas**
Currently allows for three choices for parameter uncertainty estimation: robust sandwich matrix, cross-product gradient matrix, and the observed Fisher information matrix

### CODE REVIEW

File: src/pharmpy/model/external/nonmem/update.py [hunk 1]
```diff
@@ -1473,7 +1473,13 @@ def update_estimation(control_stream, model):
...
         last_est_rec = control_stream.get_records('ESTIMATION')[-1]
         idx_cov = control_stream.records.index(last_est_rec)
-        covrec = create_record('$COVARIANCE\n')
+        if new_cov == 'SANDWICH':
+            covrec_ = '$COVARIANCE'
+        elif new_cov == 'CPG':
+            covrec_ = '$COVARIANCE MATRIX=S'
+        elif new_cov == 'OFIM':
+            covrec_ = '$COVARIANCE MATRIX=R'
+        covrec = create_record(f'{covrec_}\n')
```
Comment (stellabelin) [inline src/pharmpy/model/external/nonmem/update.py:1482]: Could be good to have some code that tests this new logic, e.g. in this test: https://github.com/pharmpy/pharmpy/blob/60ec348d5e52f7f0a9f29587d99999dc0023a489/tests/nonmem/test_nonmem_model.py#L682

File: tests/modeling/test_estimation_steps.py [hunk 2]
```diff
@@ -88,14 +88,14 @@ def test_remove_estimation_step(testdata, load_model_for_test):
 def test_add_covariance_step(testdata, load_model_for_test):
```
Comment (stellabelin) [inline tests/modeling/test_estimation_steps.py:88]: In this test I would also add a test for at least one of the new options.

**Review by stellabelin — COMMENTED** [review]
Overall looks good! But merge conflict needs to be resolved and I think it would be good to add a couple of tests.

File: tests/modeling/test_estimation_steps.py (hunk 2 above)
Comment (vrognas) [reply tests/modeling/test_estimation_steps.py:88]: Added for MATRIX=S

File: src/pharmpy/model/external/nonmem/update.py (hunk 1 above)
Comment (vrognas) [reply src/pharmpy/model/external/nonmem/update.py:1482]: Added: https://github.com/pharmpy/pharmpy/blob/c4c5dae3950ff7fba6e4cbac2d4d6895f41afdc8/tests/nonmem/test_nonmem_model.py#L658-L659

**Comment by stellabelin** [comment]
LGTM!

**Pull request merged**